frontend/src/metabase-types/api/dataset.ts

```diff
@@ -43,14 +43,22 @@ export interface ResultsMetadata {
   columns: DatasetColumn[];
 }
 
+export type Insight = {
+  col: string;
+  unit: DatetimeUnit;
+  [key: string]: any;
+};
+
 export interface DatasetData {
   rows: RowValues[];
   cols: DatasetColumn[];
+  insights?: Insight[];
+  results_metadata: ResultsMetadata;
   rows_truncated: number;
   requested_timezone?: string;
   results_timezone?: string;
   download_perms?: DownloadPermission;
-  results_metadata: ResultsMetadata;
+  native_form?: any;
```

Comment:
Can we use a real type here? Not sure about `params` though

```ts
native_form: { query: string }
```
<img width="466" alt="CleanShot 2023-12-13 at 17 19 44@2x" src="https://github.com/metabase/metabase/assets/17258145/65588842-169a-41de-8e64-4b28090b61bf">

Reply:
I cannot find a place where `native_form` is even being used on the front-end to check in what cases `params` is used